### PR TITLE
plowing effect unit test

### PR DESCRIPTION
**Pull request by deepanshubansal01**
This PR implements unit test for plowing effect feature, more details here  #3164

### CODE REVIEW

File: test/integration/plowing_effect.cc [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      {
+        EXPECT_EQ(angle, 0);
+      }
+
+      // plowing effect
+      else if(wheelSpeedLongitudinal > deadbandVelocity_ && wheelSpeedLongitudinal < saturationVelocity_)
+      {
+          // angle less than maxRadians_
+          ASSERT_LT(angle, maxRadians_);
+        // TODO: should we also check for exact angle values here
```
Comment (deepanshubansal01) [inline test/integration/plowing_effect.cc]: @adityapande-1995 for this should I compute the exact angle using the same logic used in ODEPhysics?

Comment (adityapande-1995) [reply test/integration/plowing_effect.cc]: We could calculate the angle and use EXPECT_NEAR macro to test, by including some tolerance.

File: test/worlds/plowing_effect_test.world [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+    <model name="sphere1">
+      <link name='base_link'>
+        <pose>0 0 0 0 0 0</pose>
+        <inertial>
+          <mass>0.5</mass>
+          <inertia>
+            <ixx>0.22799999999999998</ixx>
+            <iyy>0.7435210984814149</iyy>
+            <izz>0.9655210984814149</izz>
```
Comment (scpeters) [inline test/worlds/plowing_effect_test.world]: I would expect the `ixx`, `iyy` and `izz` values to be equal for a sphere

**Review by scpeters — COMMENTED** [review]
thanks, I will try testing this. In the meantime I noticed something odd in the sphere inertias in the test world

File: test/worlds/plowing_effect_test.world (hunk 2 above)
Comment (deepanshubansal01) [reply test/worlds/plowing_effect_test.world]: Thanks, it's corrected now.

**Comment by deepanshubansal01** [comment]
@scpeters @adityapande-1995 I have fixed the issue with model paths. The unit tests should run now without exporting the `GAZEBO_MODEL_PATH` env variable.

**Comment by scpeters** [comment]
thanks, the tests are looking great! I'm going to make a few changes, but first I'll merge this into #3164

**Review by scpeters — APPROVED** [review]

**Comment by deepanshubansal01** [comment]
> thanks, the tests are looking great! I'm going to make a few changes, but first I'll merge this into #3164

Awesome, thanks for the review:)

**Pull request merged**